src/main.rs

```diff
@@ -9,69 +9,166 @@ mod utils;
 #[cfg(test)]
 pub mod tests;
 
-use crate::compiler::AmberCompiler;
-use clap::Parser;
+use crate::compiler::{AmberCompiler, CompilerOptions};
+use clap::{Args, CommandFactory, Parser, Subcommand};
+use clap_complete::Shell;
 use colored::Colorize;
 use heraclitus_compiler::prelude::*;
 use std::error::Error;
-use std::fs;
 use std::io::{prelude::*, stdin};
-use std::path::{Path, PathBuf};
+use std::path::PathBuf;
 use std::process::Command;
+use std::{fs, io};
 
 #[derive(Parser, Clone, Debug)]
 #[command(version, arg_required_else_help(true))]
-#[derive(Default)]
-pub struct Cli {
-    #[arg(help = "'-' to read from stdin")]
+struct Cli {
+    #[command(subcommand)]
+    command: Option<CommandKind>,
```

Comment:
If we make `command` optional, and duplicate the `input` filename and `args` array at the top level, we can enter either `amber run script.ab` or `amber script.ab`.  The latter is required to run scripts with a shebang line as `./script.ab`.